src/toil/wdl/wdl_functions.py

```diff
@@ -342,16 +342,16 @@ def abspath_file(f: Any, cwd: str):
         raise WDLRuntimeError('Error processing file: ({}) of type: ({}).'.format(str(f), str(type(f))))
 
 
-def read_single_file(f: WDLFile, tempDir, fileStore, docker=False) -> str:
+def read_single_file(f: WDLFile, tempDir: str, fileStore: AbstractFileStore, docker: bool = False) -> str:
     import os
     try:
-        fpath = fileStore.readGlobalFile(f.file_path, userPath=os.path.join(tempDir, f.file_name))
+        fpath = fileStore.readGlobalFile(f.file_path, userPath=os.path.join(tempDir, cast(str, f.file_name)))
     except:
-        fpath = os.path.join(tempDir, f.file_name)
-    return fpath
+        fpath = os.path.join(tempDir, cast(str, f.file_name))
```

Comment:
Better to add the types of the attributes to class WDLFile than to repetitively cast in multiple places.